### PR TITLE
Added DataTypes

### PR DESCRIPTION
**Pull request by IamFaizanKhalid**

### CODE REVIEW

File: float64.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package nulltypes
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+)
+
+//NullFloat64 is a wrapper around float64
```
Comment (fahadsiddiqui) [inline float64.go]: Fix comment.

File: bool.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	Bool  bool
+	Valid bool
+}
+
+// Bool method to get NullBool object from bool
+func Bool(Bool bool) NullBool {
+	return NullBool{Bool, true}
+}
+
+//MarshalJSON method is called by json.Marshal,
```
Comment (fahadsiddiqui) [inline bool.go]: Comments not properly formatted 

```suggestion
// MarshalJSON method is called by json.Marshal,
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module github.com/datumbrain/nulltypes
+
+go 1.15
```
Comment (fahadsiddiqui) [inline go.mod:3]: Does it require any third party package?

File: bool.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+//NullBool is a wrapper around bool
+type NullBool struct {
+	Bool  bool
+	Valid bool
+}
+
+// Bool method to get NullBool object from bool
+func Bool(Bool bool) NullBool {
+	return NullBool{Bool, true}
```
Comment (fahadsiddiqui) [inline bool.go:16]: It can be a named assignment

```
NullBool{x: _, y: _}
```

File: bool.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	rt, ok := value.(bool)
+	if ok {
+		*x = NullBool{rt, true}
+	} else {
+		*x = NullBool{rt, false}
+	}
+	return nil
+}
+
+// satifies the driver.Value interface
```
Comment (fahadsiddiqui) [inline bool.go:50]: The first word of comment should be a function name.

**Review by fahadsiddiqui — CHANGES_REQUESTED** [review]
In addition to all requested changed, `go fmt ./...` on root and push it.

File: go.mod (hunk 3 above)
Comment (IamFaizanKhalid) [reply go.mod:3]: No. It only uses `database/sql/driver`, `encoding/json` and `time`.

File: bool.go (hunk 4 above)
Comment (IamFaizanKhalid) [reply bool.go:16]: It can be. But no need to put extra code here. We should make it even shorter than that if we can. 😉

File: bool.go (hunk 5 above)
Comment (IamFaizanKhalid) [reply bool.go:50]: These are not for the user to call, these are just implementing an interface.

**Review by fahadsiddiqui — APPROVED** [review]

**Pull request merged**